### PR TITLE
IterableAsList

### PR DESCRIPTION
**Pull request by g4s8**
`IterableAsList` to make a list from iterable.

### CODE REVIEW

**Comment by 0crat** [comment]
@g4s8 I'm not managing this repo, remove the [webhook](https://github.com/yegor256/cactoos/settings/hooks) or contact me in [Slack](http://www.zerocracy.com) //cc @yegor256

File: src/main/java/org/cactoos/list/IterableAsList.java [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+    /**
+     * Ctor.
+     *
+     * @param iterable An {@link Iterable}
+     */
+    public IterableAsList(final Iterable<T> iterable) {
+        super();
+        this.origin = Collections.list(
+            new IteratorAsEnumeration<>(
+                iterable.iterator()
```
Comment (yegor256) [inline src/main/java/org/cactoos/list/IterableAsList.java]: @g4s8 this is not acceptable. We can't have any code inside ctors. You're calling `.iterator()`

**Comment by yegor256** [comment]
@g4s8 good idea, but see my comment above

**Comment by yegor256** [comment]
@g4s8 also, I would put that `Repeat` into a different PR

File: src/test/java/org/cactoos/list/IterableAsListTest.java [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+     *
+     * @throws Exception if failed
+     */
+    @Test
+    public void elementAtIndexTest() throws Exception {
+        // @checkstyle MagicNumber (7 lines)
+        MatcherAssert.assertThat(
+            new IterableAsList<>(
+                new ArrayAsIterable<>(
+                    1, 2, 3, 4
```
Comment (yegor256) [inline src/test/java/org/cactoos/list/IterableAsListTest.java]: @g4s8 it's not obvious that number `4` in the array must be exactly the same as the number `4` two lines below. It's better to introduce a variable for them.

File: src/test/java/org/cactoos/list/IterableAsListTest.java [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+     * Test size of list.
+     *
+     * @throws Exception if failed
+     */
+    @Test
+    public void sizeTest() throws Exception {
+        // @checkstyle MagicNumber (5 lines)
+        MatcherAssert.assertThat(
+            new IterableAsList<>(
+                Collections.nCopies(42, 0)
```
Comment (yegor256) [inline src/test/java/org/cactoos/list/IterableAsListTest.java]: @g4s8 same here

**Comment by yegor256** [comment]
@g4s8 a few comments above

File: src/main/java/org/cactoos/list/IterableAsList.java [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+        this.source = iterable;
+        this.cache = new ArrayList<>(0);
+        this.length = new LengthOfIterable(iterable);
+    }
+
+    @Override
+    public T get(final int index) {
+        if (index < 0 || index >= this.size()) {
+            throw new IndexOutOfBoundsException(
+                String.format(
```
Comment (DronMDF) [inline src/main/java/org/cactoos/list/IterableAsList.java]: Maybe it's better to use cactoos.text.FormattedText?

**Comment by g4s8** [comment]
@yegor256, @DronMDF All done!

File: src/test/java/org/cactoos/list/IterableAsListTest.java [hunk 5]
```diff
@@ -0,0 +1,112 @@
...
+     *
+     * @throws Exception if failed
+     */
+    @Test
+    public void elementAtIndexTest() throws Exception {
+        final int expected = 4;
+        final int position = 3;
+        final Integer[] array = Collections.nCopies(position + 1, 0)
+            .toArray(new Integer[0]);
+        array[position] = expected;
```
Comment (yegor256) [inline src/test/java/org/cactoos/list/IterableAsListTest.java]: @g4s8 we are trying to make all tests single statement. There should be nothing but variables declaration and `assertThat`. I would refactor it like this:

```java
int num = 345;
int[] array = {0, 1, 2, num, 3, 4};
assertThat(IterableAsList(...).get(3), equalTo(num))
```

**Comment by yegor256** [comment]
@g4s8 see above

**Comment by g4s8** [comment]
@yegor256 fixed

**Comment by yegor256** [comment]
@rultor merge

**Pull request merged**